plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/TestHiveThriftMetastoreWithS3.java

```diff
@@ -0,0 +1,174 @@
+        this.awsAccessKey = requireNonNull(awsAccessKey, "awsAccessKey is null");
+        this.awsSecretKey = requireNonNull(awsSecretKey, "awsSecretKey is null");
+        this.writableBucket = requireNonNull(writableBucket, "writableBucket is null");
+        this.schemaName = "test_thrift_s3_" + randomTableSuffix();
+
+        String coreSiteXmlContent = Resources.toString(Resources.getResource("s3/hive-core-site.template.xml"), UTF_8)
+                .replace("%S3_BUCKET_ENDPOINT%", s3endpoint)
+                .replace("%AWS_ACCESS_KEY_ID%", awsAccessKey)
+                .replace("%AWS_SECRET_ACCESS_KEY%", awsSecretKey);
+
+        hadoopCoreSiteXmlTempFile = Files.createTempFile("core-site", ".xml", PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rw-r--r--")));
+        hadoopCoreSiteXmlTempFile.toFile().deleteOnExit();
+        Files.writeString(hadoopCoreSiteXmlTempFile, coreSiteXmlContent);
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        HiveHadoop hiveHadoop = HiveHadoop.builder()
+                .withFilesToMount(ImmutableMap.of("/etc/hadoop/conf/core-site.xml", hadoopCoreSiteXmlTempFile.normalize().toAbsolutePath().toString()))
+                .build();
+        hiveHadoop.start();
+
+        return S3HiveQueryRunner.builder()
+                .setHiveMetastoreEndpoint(hiveHadoop.getHiveMetastoreEndpoint())
+                .setS3Endpoint(s3endpoint)
+                .setS3AccessKey(awsAccessKey)
+                .setS3SecretKey(awsSecretKey)
+                .setBucketName(writableBucket)
+                .setPopulateTpchData(false)
+                .setThriftMetastoreConfig(new ThriftMetastoreConfig().setDeleteFilesOnDrop(true))
+                .setHiveProperties(ImmutableMap.of("hive.allow-register-partition-procedure", "true"))
+                .build();
+    }
+
+    @BeforeClass
+    public void setUp()
+    {
+        String schemaLocation = "s3a://%s/%s".formatted(writableBucket, schemaName);
+        assertUpdate("CREATE SCHEMA " + schemaName + " WITH (location = '" + schemaLocation + "')");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        assertUpdate("DROP SCHEMA IF EXISTS " + schemaName);
+    }
+
+    @Test
+    public void testRecreateTable()
+    {
+        String tableName = "%s.test_recreate_table_%s".formatted(schemaName, randomTableSuffix());
+
+        assertUpdate("CREATE TABLE " + tableName + "(col int)");
+        try {
+            assertUpdate("INSERT INTO " + tableName + " VALUES (1)", 1);
+            assertUpdate("DROP TABLE " + tableName);
+
+            // Recreating a table throws "Target directory for table 'xxx' already exists" in real S3 (not MinIO)
+            // when 'hive.metastore.thrift.delete-files-on-drop' config property is false
+            assertUpdate("CREATE TABLE " + tableName + "(col int)");
```

Comment:
Maybe it would make sense to check that the table in fact is empty?

Reply:
Included additional assertions.